auth/src/main/java/software/amazon/awssdk/auth/signer/internal/Aws4SignerRequestParams.java

```diff
@@ -80,12 +76,10 @@ public Aws4SignerRequestParams(Aws4SignerParams signerParams) {
     /**
      * Returns the signing date from the request.
      */
-    private long getSigningDate(Integer timeOffset) {
-        if (timeOffset == null) {
-            return System.currentTimeMillis();
-        } else {
-            return System.currentTimeMillis() - timeOffset * 1000L;
-        }
+    private long getSigningDate(Optional<Integer> timeOffset) {
+        return timeOffset.map(t -> System.currentTimeMillis() - t * 1000L)
+                         .orElse(System.currentTimeMillis());
+
```

Comment:
Can we wrap it in optional internally instead of taking optional as a param?

Reply:
Fixed. Updated it to take `Aws4SignerParams` as a parameter.